### PR TITLE
chore: introduce and use `TargetRuntime`

### PR DESCRIPTION
**Pull request by mogery**
**Summary:**  
Adds `TargetRuntime` struct and uses it for `ConfigReader` and `AppContext` to simplify piping.

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/http/request_context.rs [hunk 1]
```diff
@@ -102,18 +102,18 @@ impl RequestContext {
...
             req_headers: HeaderMap::new(),
             http_data_loaders: server_ctx.http_data_loaders.clone(),
             gql_data_loaders: server_ctx.gql_data_loaders.clone(),
-            cache: server_ctx.cache.clone(),
+            cache: server_ctx.runtime.cache.clone(),
             grpc_data_loaders: server_ctx.grpc_data_loaders.clone(),
             min_max_age: Arc::new(Mutex::new(None)),
             cache_public: Arc::new(Mutex::new(None)),
-            env_vars: server_ctx.env_vars.clone(),
+            env_vars: server_ctx.runtime.env.clone(),
```
Comment (tusharmath) [inline src/http/request_context.rs:116]: There is so much of copying here, we can just refer to `server_ctx` directly instead. Happy to accept this via a separate PR.

File: src/target_runtime.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+use std::sync::Arc;
+
+use async_graphql_value::ConstValue;
+
+use crate::{Cache, EnvIO, FileIO, HttpIO};
+
+#[derive(Clone)]
+pub struct TargetRuntime {
```
Comment (tusharmath) [inline src/target_runtime.rs:8]: Some doc about it's relevance would be very helpful!

File: src/http/request_context.rs (hunk 1 above)
Comment (mogery) [reply src/http/request_context.rs:116]: added to my todo list :)

File: tests/graphql_spec.rs [hunk 3]
```diff
@@ -335,10 +336,13 @@ async fn test_execution() -> std::io::Result<()> {
...
-                    h2_client,
-                    init_env(),
-                    Arc::new(chrono_cache),
+                    TargetRuntime {
+                        http: h_client,
+                        http2_only: h2_client,
+                        env: init_env(),
+                        cache: Arc::new(chrono_cache),
+                        file: init_file(),
+                    },
```
Comment (tusharmath) [inline tests/graphql_spec.rs:345]: Can't we use `init_runtime` here?

File: src/target_runtime.rs (hunk 2 above)
Comment (mogery) [reply src/target_runtime.rs:11]: Added.

File: tests/graphql_spec.rs (hunk 3 above)
Comment (mogery) [reply tests/graphql_spec.rs]: Fixed. I thought `h_client` and `h2_client` had some special config there, but nope.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**